### PR TITLE
Glt 1288 hibernate pools

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractDilution.java [hunk 1]
```diff
@@ -59,10 +63,23 @@
   private String identificationBarcode;
   private String dilutionUserName;
   private Long preMigrationId;
-  // TODO: Create an interface for this when Hibernating
+
+  @ManyToMany(targetEntity=PoolImpl.class)
+  @JoinTable(name = "Pool_Dilution", joinColumns = { @JoinColumn(name = "dilution_dilutionId") }, inverseJoinColumns = {
+      @JoinColumn(name = "pool_poolId") })
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractDilution.java]: join columns...

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolImpl.java [hunk 2]
```diff
@@ -297,7 +295,13 @@ public String getLabelText() {
...
   @Override
   public Date getLastModified() {
-    return lastModified;
+    Date max = new Date(0);
+    for (ChangeLog c : changeLog) {
+      if (c.getTime().after(max)) {
+        max = c.getTime();
+      }
+    }
+    return max;
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolImpl.java]: One of the main ways this attribute is used is as the `sortBy` column on the list pools page, so it gets used as the `ORDER BY` column when selecting pools. Will this implementation of `getLastModified` be faster than the XDerivedInfo method for returning that query (aka should we be changing this for Samples, Libraries and Runs as well)?

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolQCImpl.java [hunk 3]
```diff
@@ -49,6 +50,7 @@
   public static final String UNITS = "nM";
 
   @ManyToOne(targetEntity = PoolImpl.class)
+  @Column(name = "pool_poolId")
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolQCImpl.java]: I think this should be `@JoinColumn`

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolImpl.java (hunk 2 above)
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolImpl.java]: I don't know. It's probably comparable. Let's consider this an experiment.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolQCImpl.java (hunk 3 above)
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolQCImpl.java]: Fixed.

File: sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernatePoolQCDao.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+import uk.ac.bbsrc.tgac.miso.core.data.type.QcType;
+import uk.ac.bbsrc.tgac.miso.core.store.PoolQcStore;
+
+/**
+ * uk.ac.bbsrc.tgac.miso.sqlstore
+ * <p/>
+ * Info
+ *
+ * @author Rob Davey
+ * @since 0.1.9
```
Comment (callunity) [inline sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernatePoolQCDao.java]: Please remove or fix @author and @since

**Comment by apmasell** [comment]
Fixed.

File: core/.settings/org.eclipse.core.resources.prefs [hunk 5]
```diff
@@ -3,5 +3,4 @@ encoding//src/main/java=UTF-8
 encoding//src/main/resources=UTF-8
 encoding//src/test/java=UTF-8
 encoding//src/test/resources=UTF-8
-encoding//target/generated-sources/java-templates=UTF-8
```
Comment (callunity) [inline core/.settings/org.eclipse.core.resources.prefs]: Was this supposed to be removed?

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolImpl.java [hunk 6]
```diff
@@ -121,6 +122,10 @@
   @ManyToOne(targetEntity = UserImpl.class)
   private User lastModifier;
 
+  @OneToOne(targetEntity = PoolDerivedInfo.class)
+  @PrimaryKeyJoinColumn
+  private PoolDerivedInfo derivedInfo;
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolImpl.java]: Where is the PoolDerivedInfo class?

Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolImpl.java]: Sitting in my working directory. >.>

File: core/.settings/org.eclipse.core.resources.prefs (hunk 5 above)
Comment (apmasell) [inline core/.settings/org.eclipse.core.resources.prefs]: No. fixed.

Comment (callunity) [inline core/.settings/org.eclipse.core.resources.prefs]: Not yet fixed...

Comment (apmasell) [inline core/.settings/org.eclipse.core.resources.prefs]: Pushed.

**Pull request merged**